src/sqlfluff/core/parser/grammar/delimited.py

```diff
@@ -151,15 +151,18 @@ def match(
                         break
 
                 with parse_context.deeper_match() as ctx:
+                    terminators = (
+                        delimiter_matchers if elements != delimiter_matchers else None
+                    )
+                    if terminators:
+                        ctx.push_terminators(terminators)
                     match, _ = self._longest_trimmed_match(
                         segments=seg_content,
                         matchers=elements,
                         parse_context=ctx,
                         # We've already trimmed
                         trim_noncode=False,
-                        terminators=delimiter_matchers
-                        if elements != delimiter_matchers
-                        else None,
+                        terminators=terminators,
```

Comment:
should this not be `ctx.terminators`?

Reply:
No - but you're right that there is duplication. In fact we shouldn't need to specify the terminators here _at all_ because they're fetched from the context within that function anyway. 👍 

Will tidy up